bellman/src/lib.rs

```diff
@@ -356,6 +353,30 @@ impl fmt::Display for SynthesisError {
     }
 }
 
+/// An error during verification.
+#[derive(Debug)]
+pub enum VerificationError {
+    /// Verification was attempted with a malformed verifying key.
+    MalformedVerifyingKey,
```

Comment:
I copied this name from the `SynthesisError` variant but I don't like it.  It occurs when the verifying key has a different length from the number of public inputs to the proof statement.  But this does not necessarily indicate that the pvk is malformed, only that there's a mismatch.

Reply:
`VerifyingKeyMismatch`?

Reply:
`InvalidVerifyingKey` would do.

Reply:
Updated in a154154